include/openPMD/backend/Attribute.hpp

```diff
@@ -61,4 +62,81 @@ using Attribute = auxiliary::Variant< Datatype,
                             std::vector< std::string >,
                             std::array< double, 7 >,
                             bool >;
-} // openPMD
+namespace detail
+{
+    template<bool wasFound, typename U, typename T_Head, typename ...T>
+    struct FindTypeImpl
+    {
+        static constexpr bool value = std::is_same<
```

Comment:
hm, on the platforms with trouble, `is_same` will still return false although they are the same... Argh!

Reply:
turns out they are not the same. each of them are distinct type, just some happen to implement the same precision.